Autopilot/PathManager/Inc/pathStateClasses.hpp

```diff
@@ -11,6 +11,7 @@
 #include "CommsWithTelemetry.hpp"
 #include "SensorFusion.hpp"
 #include "AttitudePathInterface.hpp"
+#include "Interchip_A.h"
```

Comment:
By including Interchip_A.h, we are including a hal file which are impossible to unit test. So here is what I will recommend you do. 

Create a new HPP file in `.Autopilot/Inc` where you declare two method:
1. getPitchPercent
2. getYawPercent

Both will return a float. You will include this HPP file in `pathStateClasses.hpp` instead of `Interchip_A.h`. You  will call the functions above in the commsWithTelemetry::execute() function:

```cpp
void commsWithTelemetry::execute(pathManager* pathMgr)
 {
     GetTelemetryCommands(&_incomingData);

    static float gimbalPercentPitch = getPitchPercent(_incomingData.gimbalPitch);
    static float gimbalPercentYaw = getYawPercent(_incomingData.gimbalYaw);

     Interchip_SetPWM(3, gimbalPercentPitch);
     Interchip_SetPWM(7, gimbalPercentYaw);
```

Now you will make a cpp file in `./Autopilot/Src` that will have define the functions declared in your hpp file:

```cpp
#include "Interchip_A.h" // Include the interchip file here

// incoming angles are radians
static float gimbalMaxPitch = 3.14159;
 static float gimbalMaxYaw = 3.14159/2;

float getPitchPercent(float command) {
   // calculations you did previously

// return percentage
}

float getYawPercent(float command) {
   // calculations you did previously

// return percentage
}
```

Reply:
Note that this will cause linker problems in the Pathmanager FSM tests. So you will need to make mock functions for `getPitchPercent()` and `getYawPercent()`

I can show you how to do that when the time comes